hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -180,8 +174,8 @@ public boolean isSame(Object x, Object y) throws HibernateException {
 			return true;
 		}
 		// null value and empty component are considered equivalent
-		final Object[] xvalues = getPropertyValues( x );
-		final Object[] yvalues = getPropertyValues( y );
+		final var xvalues = getPropertyValues( x );
+		final var yvalues = getPropertyValues( y );
```

Comment:
var makes the code unique and simple, nice effort taken.

Might be related to this one, as its kind of challenge to maintain a convention like this on scale.

- https://docs.openrewrite.org/recipes/java/migrate/lang/var/usevarforobject